src/main/java/capstone/mfslbackend/service/FantasyLeagueService.java

```diff
@@ -27,4 +31,18 @@ public List<FantasyLeague> getFantasyLeagueByName(String fantasyLeagueName) {
         String name = "%" + fantasyLeagueName + "%";
         return fantasyLeagueRepository.findFantasyLeagueByLeagueNameLikeIgnoreCase(name);
     }
-}
+
+    public FantasyLeague joinFantasyLeague(String username, Long leagueId) {
+            Optional<FantasyLeague> fantasyLeagueOptional = getFantasyLeagueById(leagueId);
+            if (fantasyLeagueOptional.isEmpty()) {
+                log.error("Fantasy League with id {} not found", leagueId);
+                throw new IllegalArgumentException("Fantasy League with id " + leagueId + " not found");
+            }
+
+            FantasyLeague fantasyLeague = fantasyLeagueOptional.get();
+            User user = userService.getUser(username);
```

Comment:
You don't do an error check if the user doesn't exist

Reply:
Oh yes, forgot that the User repository doesn't return an optional. That wasn't implemented in the getUser either.

@linamoufarreh check if the returned user is null, if it is then throw an error. Otherwise its good.